packages/owl-bot/src/owl-bot.ts

```diff
@@ -49,11 +50,23 @@ interface PubSubContext {
   };
 }
 
-export function OwlBot(
-  privateKey: string | undefined,
-  app: Probot,
-  db?: Db
-): void {
+async function acquireLock(target: string): Promise<DatastoreLock | boolean> {
```

Comment:
The bulk of the logic is here, double check for me, goal:

1. if we a lock already exists, return false immediately.
2. otherwise acquire and return lock.